cylc/uiserver/tests/test_uiserver.py

```diff
@@ -18,101 +18,77 @@
+    mock_config(CylcUIServer={'ui_build_dir': 'beef'})
+    uis = CylcUIServer()
     with pytest.raises(Exception):
-        uis = CylcUIServer(8000, '/wellington/')
+        uis.initialize_settings()
         uis.ui_path  # forces computation of the ui_path
 
 
 def test_ui_version_specified(mock_config, ui_build_dir):
     """If the UI version is specified it should use that one."""
     # test with an installed version
-    mock_config(f'''
-        c.UIServer.ui_build_dir="{ui_build_dir}"
-        c.UIServer.ui_version="1.0"
-    ''')
-    uis = CylcUIServer(8000, '/prefix/')
+    mock_config(CylcUIServer={
+        'ui_build_dir': ui_build_dir,
+        'ui_version': '1.0'
+        })
+    uis = CylcUIServer()
+    uis.initialize_settings()
     assert uis.ui_path == ui_build_dir / '1.0'
 
     # test with an uninstalled version
-    mock_config(f'''
-        c.UIServer.ui_build_dir="{ui_build_dir}"
-        c.UIServer.ui_version="5.0"
-    ''')
+    mock_config(CylcUIServer={
+        'ui_build_dir': ui_build_dir,
+        'ui_version': '5.0'
+        })
+    uis = CylcUIServer()
     with pytest.raises(Exception):
-        uis = CylcUIServer(8000, '/prefix/')
+        uis.initialize_settings()
         uis.ui_path  # forces computation of the ui_path
 
 
 def test_ui_build_dir(mock_config, ui_build_dir):
     """If the UI build path does is a UI build it should use it."""
-    mock_config(f'''
-        c.UIServer.ui_build_dir="{ui_build_dir / '2.0'}"
+    mock_config(CylcUIServer={
+        'ui_build_dir': ui_build_dir / '2.0',
         # the ui_version should do nothing here
-        c.UIServer.ui_version="99"
-    ''')
-    uis = CylcUIServer(8000, '/prefix/')
+        'ui_version': '99'
+        })
+    uis = CylcUIServer()
+    uis.initialize_settings()
     assert uis.ui_path == ui_build_dir / '2.0'
 
 
-def test_ui_build_dir_argument(mock_config, ui_build_dir):
-    """The ui_build_dir argument should override any config value."""
-    # config asks for 2.0
-    mock_config(f'''
-        c.UIServer.ui_build_dir="{ui_build_dir / '2.0'}"
-    ''')
-    uis = CylcUIServer(8000, '/prefix/')
-    # normally this would be used
-    assert uis.ui_path == ui_build_dir / '2.0'
-    # argument asks for 1.0
-    uis = CylcUIServer(8000, '/prefix/', ui_build_dir / '1.0')
-    # argument wins
-    assert uis.ui_path == ui_build_dir / '1.0'
+# def test_ui_build_dir_argument(mock_config, ui_build_dir):
+#     """The ui_build_dir argument should override any config value."""
+#     # config asks for 2.0
+#     mock_config(CylcUIServer={
+#         'ui_build_dir': ui_build_dir / '2.0'
+#         })
+#     uis = CylcUIServer()
+#     uis.initialize_settings()
+#     # normally this would be used
+#     assert uis.ui_path == ui_build_dir / '2.0'
+#     # argument asks for 1.0
+#     uis = CylcUIServer(8000, '/prefix/', ui_build_dir / '1.0')
+#     uis.initialize_settings()
+#     # argument wins
+#     assert uis.ui_path == ui_build_dir / '1.0'
```

Comment:
Build dir argument has been removed, you can still use environment variables + logic in your `jupyter_config` file.

Reply:
Update, CLI args for jupyter_server extensions were broken but have since been fixed in version 1.10.2, however, we can't use that yet due to the issue of detecting token authenticated connections (see caveats in issue description above).

Reply:
Bumped to https://github.com/cylc/cylc-uiserver/issues/235